net: http: service: check for uncompressed file correctly

AndreyDodonov-EH opened the pull request:
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/92917

Correct not resetting compression extension,
when checking for an uncompressed file,
in case if compression is enabled, but no compressed file found.

So  if we have `CONFIG_HTTP_SERVER_COMPRESSION=y`, but for a specific file no compressed version is found in the filesystem, instead for checking for e.g. `index.html`, we will be checking for the last checked compressed version, `index.html.zz`

This happens because extension is not cleared.
Correct that.

mrodgers-witekio commented on subsys/net/lib/http/http_server_core.c:
```diff
@@ -847,6 +847,8 @@ int http_server_find_file(char *fname, size_t fname_size, size_t *file_size,
 			}
 		}
 	}
+	/* No compressed file found, try the original filename */
```
I think the PR is fine as it is - the `strlen` outside the block was there already, and is optimised out by the compiler anyway (I just built the sample app without compression enabled and took a quick look at the disassembly)

But if you want to clean this up, I would probably lean towards declaring `len` inside the block too unless Robert/Jukk disagree - there are already other instances in the same file where it's done this way, e.g. see line 170

AndreyDodonov-EH committed:
net: http: service: check for uncompressed file correctly

Correct not resetting compression extension,
when checking for an uncompressed file,
in case if compression is enabled, but no compressed file found.

Signed-off-by: Andrey Dodonov <Andrey.Dodonov@endress.com>
